book/src/concepts/casting.md

```diff
@@ -0,0 +1,39 @@
+<!--
+SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+SPDX-FileContributor: Ben Ford <ben.ford@kdab.com>
+
+SPDX-License-Identifier: MIT OR Apache-2.0
+-->
+
+# Casting
```

Comment:
wonder if we note somewhere how this only is implemented for RustQt types for now that have qobject attribute and optionally the base attribute.

As in a future PR support will be added for C++Qt (hopefully the next PR ;-) )